src/ray/raylet/scheduling/scheduling_policy.cc

```diff
@@ -204,5 +204,37 @@ int64_t SchedulingPolicy::HybridPolicy(const ResourceRequest &resource_request,
                                 require_available, is_node_available);
 }
 
+int64_t SchedulingPolicy::RandomPolicy(const ResourceRequest &resource_request,
```

Comment:
There was a previous discussion about how we should handle 0 cpu actors (we should consider it in the scheduler refactor): basically we should let gcs decides the random node to send lease request and the raylet should just grant_or_reject instead of now having (N-1)/N chance of spillback.

Reply:
> basically we should let gcs decides the random node to send lease request ...

If we let the per raylet distributed scheduler to randomly pick a node (and don't allow the picked node to further spill), it is equivalent to gcs based solution?

Reply:
I think if gcs also runs the distributed scheduler (which I think it will based on the new design?), then it will randomly pick a raylet. That raylet only runs the local scheduler to grant or reject.

Reply:
I believe this codepath is only for 0 CPU actor (the default). So for actors that require resources, they would use the different code path. Just for this path, it'd be similar to GCS based scheduler